Sources/MapLibreSwiftUI/MapViewCoordinator.swift

```diff
@@ -128,20 +128,11 @@ public class MapViewCoordinator<T: MapViewHostViewController>: NSObject, MLNMapV
             mapView.minimumPitch = pitchRange.rangeValue.lowerBound
             mapView.maximumPitch = pitchRange.rangeValue.upperBound
         case let .trackingUserLocationWithHeading(zoom: zoom, pitch: pitch, pitchRange: pitchRange):
-            mapView.userTrackingMode = .followWithHeading
+            mapView.setUserTrackingMode(.followWithHeading, animated: animated) {
+                mapView.minimumPitch = pitchRange.rangeValue.lowerBound
+                mapView.maximumPitch = pitchRange.rangeValue.upperBound
 
-            if mapView.frame.size == .zero {
```

Comment:
This is a draft, because I haven't yet figured out what this `if mapView.frame.size == .zero` branch is actually trying to accomplish. 

I could leave it in - but if the problem it's trying to prevent was also caused by disregarding animation context, it'd be nice to get rid of it.

Reply:
Yeah, the reason this is in here is because of animations. I'll try to test this and see if maybe it's no longer necessary. I _think_ one of the examples in this repo should misbehave if this is still needed.

Reply:
I added this. The problem is that the way mln calculates camera settings, some of its values are inferre  from the frame of the mlnmapview. But during init, this .zero. So if you try to initialize our MapView with a camera that sets a pitch and zoom level, this is instead ignored - see https://github.com/maplibre/swiftui-dsl/pull/37 and https://github.com/maplibre/maplibre-native/blob/63d67079e0ab6cf73d37223a50b81c3cf7128e40/platform/darwin/src/MLNGeometry.mm#L48  - where the size is used to calculate things (and other functions in this file). 

This leads to a problem when you initialize MapView directly with a camera - this camera is incorrectly set up then. If you set up the camera a moment later (in onAppear for example) the same camera is then correctly set up as the .frame is no longer .zero.

We could of course run some code to re-set the camera on first appear instead of my .zero workaround here I guess, if my solution is causing problems.

Reply:
Ok, I had a chance to try this out in a bit more detail and it looks like some variant of the original code both here and below in the course tracking branch is needed. For a minimal example of the case @hactar described, see the course tracking example in `User Location.swift`. Previously, it would correctly transition to the "puck" but it doesn't anymore.

Reply:
Resolved after restoring the check; examples and demo apps work correctly now.